### PR TITLE
JDK-6725221 Standardize obtaining boolean properties with defaults

### PR DESCRIPTION
**Pull request by mcpowers**
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 reviewer)

### Issue
 * [JDK-6725221](https://bugs.openjdk.java.net/browse/JDK-6725221): Standardize obtaining boolean properties with defaults

### Reviewers
 * [Phil Race](https://openjdk.java.net/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to [ee435c4d](https://git.openjdk.java.net/jdk/pull/8559/files/ee435c4d5faf49f66ae7de1a55ad84eba3135f30)
 * [Roger Riggs](https://openjdk.java.net/census#rriggs) (@RogerRiggs - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8559/head:pull/8559` \
`$ git checkout pull/8559`

Update a local copy of the PR: \
`$ git checkout pull/8559` \
`$ git pull https://git.openjdk.java.net/jdk pull/8559/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8559`

View PR using the GUI difftool: \
`$ git pr show -t 8559`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8559.diff">https://git.openjdk.java.net/jdk/pull/8559.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/java/lang/reflect/AccessibleObject.java [hunk 1]
```diff
@@ -773,11 +773,8 @@ private boolean slowVerifyAccess(Class<?> caller, Class<?> memberClass,
      */
     private static boolean printStackTraceWhenAccessFails() {
         if (!printStackPropertiesSet && VM.initLevel() >= 1) {
-            String s = GetPropertyAction.privilegedGetProperty(
-                    "sun.reflect.debugModuleAccessChecks");
-            if (s != null) {
-                printStackWhenAccessFails = !s.equalsIgnoreCase("false");
-            }
+            printStackWhenAccessFails = GetBooleanAction.
+                privilegedGetProperty("sun.reflect.debugModuleAccessChecks");
```
Comment (AlanBateman) [inline src/java.base/share/classes/java/lang/reflect/AccessibleObject.java:777]: Running with `-Dsun.reflect.debugModuleAccessChecks` should set printStackWhenAccessFails to true, not false.

**Review by prrace — APPROVED** [review]
The xtoolkit change is fine. I've not looked at anything else
You'll clearly need multiple reviewers for this one.

**Comment by bradfordwetmore** [comment]
Sorry for the confusion.

The original intent of this bug 14 years ago was to standardize the seclibs code where simple getProperty calls were needed in the context of an AccessController, without having to provide the doProvided calls. i.e. GetBooleanAction.privilegedGetProperty(). This was not intended not other parts of the JDK.

For example: ChannelImpl.java provides a getBooleanProperty() method, which is very similar to GetBooleanAction. I noticed other places in the security where this was being done.

Some of the classes like Debug have been rewritten (SSLLogger), so the issue does not appear to exist there any logger.

The certpath code is working with Security.getProperty(), so that was a red herring.

**Comment by prrace** [comment]
I did wonder why it has security-libs as the sub-category and if the intent was not what we see here.

**Comment by AlanBateman** [comment]
> I did wonder why it has security-libs as the sub-category and if the intent was not what we see here.

I suspect the JBS issue was initially created to to look at the usages of getProperty in the security code but it has been extended. The issue is that it changes the meaning of the empty value case in at least two places so each one will require careful attention.

File: src/java.base/share/classes/java/lang/reflect/AccessibleObject.java (hunk 1 above)
Comment (mcpowers) [reply src/java.base/share/classes/java/lang/reflect/AccessibleObject.java:777]: You are right. The old way maps the null string to true, and the new way maps it to false. I did not notice that. At this point, I see no value in making the "true".equals and "false".equals changes. Too much can break. I'll reverse these changes.

Comment (RogerRiggs) [reply src/java.base/share/classes/java/lang/reflect/AccessibleObject.java:777]: This change still needs to be reversed.

Comment (mcpowers) [reply src/java.base/share/classes/java/lang/reflect/AccessibleObject.java]: The change has been reversed.

We had an internal discussion about IntelliJ's unboxing recommendation, and decided to let unboxing be a matter of personal preference. Those changes have been reversed too.

**Review by RogerRiggs — APPROVED** [review]
LGTM

**Comment by mcpowers** [comment]
Mach5 tier1 and tier2 completed without any failures. I don't know what to make of the pre-submit failures - lang and hotspot?

**Comment by bradfordwetmore** [comment]
> 

IIUC, these are due to Loom failures in some of the other platforms supported by OpenJDK but not by Oracle.  They are being resolved.

**Pull request closed without merging**